tztrout/data.py

```diff
@@ -90,8 +97,27 @@ def deduplicate(item):
     return deduplicate
 
 
+def generate_us_zipcode_namedtuples():
+    """Generate a list of namedtuples for US zipcodes from the list of
+    dictionaries in data/us_zipcode_data.json
+    """
+    if not os.path.exists(US_ZIPCODE_DATA_PATH):
+        raise ValueError(
+            'US Zipcode data missing. Run regenerate_data.py first'
+        )
+    us_zipcode_tuples = []
+    with open(US_ZIPCODE_DATA_PATH, 'r') as file:
+        data = json.loads(file.read())
+    for zip in data:
+        us_zipcode_tuples.append(Zipcode(**zip))
+    return us_zipcode_tuples
+
+
+ZIPCODE_TUPLES = generate_us_zipcode_namedtuples()
```

Comment:
Assuming you don't use it anywhere but `InMemoryZipData`, don't leave it here as a constant hogging extra memory - just call the function when needed down below.

Reply:
Done in 79b5bd4